components/Nav/Nav.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { render, screen } from '@testing-library/react';
+import Nav from './Nav';
+
+const mockPush = jest.fn();
+const mockUsePathname = jest.fn();
+
+const mockUseAuthContext = {
+  logoutAccount: jest.fn(),
+};
+
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      push: mockPush,
+    };
+  },
+  usePathname() {
+    return mockUsePathname();
+  },
+}));
+
+jest.mock('../../context/AuthContextProvider', () => ({
+  useAuthContext() {
+    return {
+      ...mockUseAuthContext,
+    };
+  },
+}));
+
+describe('Nav', () => {
+  beforeAll(() => {
+    jest.clearAllMocks();
+  });
+  it('it should be hidden when path is /register', () => {
+    mockUsePathname.mockImplementation(() => '/register');
+
+    render(<Nav />);
+
+    const navElement = screen.getByTestId('nav');
+
+    expect(navElement).toBeInTheDocument();
+    expect(navElement).toHaveClass('hidden');
+  });
+
+  it('it should be hidden when path is /login', () => {
+    mockUsePathname.mockImplementation(() => '/login');
+
+    render(<Nav />);
+
+    const navElement = screen.getByTestId('nav');
+
+    expect(navElement).toBeInTheDocument();
+    expect(navElement).toHaveClass('hidden');
+  });
+
+  it('it should be visible when logged in', () => {
+    render(<Nav />);
+
+    const navElement = screen.getByTestId('nav');
+
+    expect(navElement).toBeInTheDocument();
+    expect(navElement).toHaveClass('flex');
```

Comment:
M: This test is failing in your build, please fix this so the test passes as intended.  This test returns the following error:
```FAIL components/Nav/Nav.test.tsx
  ● Nav › it should be visible when logged in

    expect(element).toHaveClass("flex")

    Expected the element to have class:
      flex
    Received:
      h-[16](https://github.com/LetsGetTechnical/gridiron-survivor/actions/runs/9192687730/job/25281957267?pr=261#step:6:17) items-center border-b border-zinc-100 from-[#4E160E] to-zinc-950 px-4 dark:border-zinc-800 dark:bg-gradient-to-b hidden```

Reply:
m: This test should check that the class `hidden` is NOT existent.

Reply:
Understood, I'll update it. Thank you guys!